### DISCUSSION TITLE
Adding custom Requester into .Net DI

### DISCUSSION THREAD

**Discussion by Guram-K** [canton7/RestEase#244] [Q&A] [answered]
I'm trying to add into DI custom Requester to implement logging for my application but can't seem to find correct way of doing it. I had few attempts but it doesn't enter the overriden methods, what's the proper way of registering custom requester with overrides from Requester as a base class?

Using .Net 6 and latest version of RestEase

```
public class LoggingRequester: Requester
{
    public LoggingRequester(HttpClient httpClient) : base(httpClient) { }

    public override async Task<T> RequestAsync<T>(IRequestInfo requestInfo)
    {
        //todo: log request

        var response = await base.RequestAsync<T>(requestInfo);

        //todo: log response

        return response;
    }
}
```

**Comment by canton7** [comment]
I assume you're using HttpClientFactory? You should be able to do:

```cs
services.AddRestEaseClient<ISomeApi>("https://api.example.com")
    .AddHttpMessageHandler<LoggingRequester>()
```

Where LoggingRequester inherits from `DelegatingHandler`.

**Comment by Guram-K** [comment]
Yes I'm using HttpClientFactory. 
`AddHttpMessageHandler()` adds a `DelegatingHandler` and that's not what I'm trying to achieve, I'm trying to override existing Requester class to add extra functionality while still calling `base.RequestVoidAsync(requestInfo)`,  `base.RequestWithResponseAsync<T>(requestInfo)`and other functions, basically this is what I'm trying to do: 
![image](https://user-images.githubusercontent.com/57315757/204792513-291e855a-ead2-404b-8a79-09133d55def7.png)
but can't figure out how to provide an instance of my subclass to `RestClient.For<T>`.

**Reply by canton7** [reply]
Ah right, yeah there's no support for this with HttpClientFactory yet. I can get that in shortly...

**Reply by Guram-K** [reply]
Ok, if you could add it in feature versions would be really useful.
Thanks for help.

**Comment by canton7** [comment, marked as answer]
This will be out in v1.6.1 shortly

**Reply by Guram-K** [reply]
Just tried it out, works like a charm.
For anyone who will search for a quick answer you can register it like this:
```
services.AddRestEaseClient<ISomeApi>("https://api.example.com", new()
    {
        RequesterFactory = httpClient => new CustomRequester(httpClient)
    }
);
```
